jwst/flatfield/flat_field.py

```diff
@@ -301,34 +301,47 @@ def do_NIRSpec_flat_field(output_model,
         xstop = xstart + xsize
         ystop = ystart + ysize
 
-        # Make sure there is a WCS.
-        if not hasattr(slit.meta, "wcs") or slit.meta.wcs is None:
-            log.error("Slit %s does not have a 'wcs' attribute.", slit.name)
-            if output_model.meta.cal_step.assign_wcs == 'COMPLETE':
-                raise RuntimeError("WCS was not found, but it isn't clear "
-                                   "why not.")
+        got_wcs = hasattr(slit.meta, "wcs") and slit.meta.wcs is not None
+
+        # Get the wavelength at each pixel in the extracted slit data.
+        wl = slit.wavelength                    # a 2-D array
+        got_wl_attribute = True
+        # There must be either a wavelength array or a meta.wcs.
+        if wl.min() == 0. and wl.max() == 0.:
```

Comment:
wl.min() and wl.max() will return np.nan for Nirspec data since there are NaNs in the arrays. Is `np.nanmin()` better?

Reply:
This should be OK.  If I really wanted the min or max, I would need to exclude NaNs, but all I need to know is that the values are not zero.  If there are NaNs in the array, it has at least been populated with something.

Reply:
Ah, I see. It makes sense.